### PR TITLE
[CIT-279] Adopt user model for system test

### PR DESCRIPTION
**Pull request by abelikt**

### CODE REVIEW

File: tests/PySys/local_mqtt_pub_sub/run.py [hunk 1]
```diff
@@ -18,26 +18,37 @@
...
-            arguments=["mqtt", "sub", "atopic"],
+            command=sudo,
+            arguments=[tedge, "mqtt", "sub", "atopic"],
             stdouterr="tedge_sub",
             background=True,
         )
 
+        # Wait for a small amount of time to give tedge sub time
+        # to initialize
         time.sleep(0.1)
```
Comment (didier-wenzek) [inline tests/PySys/local_mqtt_pub_sub/run.py:32]: It's strange that this is required. `tedge mqtt sub` is doing nothing beyond a opening a socket and listening for message.

File: tests/PySys/local_mqtt_pub_sub/run.py [hunk 2]
```diff
@@ -18,26 +18,37 @@
...
-            arguments=["mqtt", "pub", "atopic", "the message"],
-            stdouterr="tedge_pub",
+            command=sudo,
+            arguments=[tedge, "mqtt", "pub", "atopic", "the message"],
+            stdouterr="tedge_pub3",
+        )
+
+        # Kill the subscriber process explicitly with sudo as PySys does
+        # not have the rights to do it
+        kill = self.startProcess(
```
Comment (didier-wenzek) [inline tests/PySys/local_mqtt_pub_sub/run.py:48]: One doesn't need to run `tedge mqtt` as sudo.

Comment (abelikt) [reply tests/PySys/local_mqtt_pub_sub/run.py:48]: I wasn't sure in this case. A strange thing that I observed there is an error when I run tedge pub with PySys:
"Error: failed to read the tedge configuration
Caused by:
User's Home Directory not found."
It works with sudo, probably PySys plays a bit around with the environment...

**Review by didier-wenzek — APPROVED** [review]
Approved.

File: tests/PySys/local_mqtt_pub_sub/run.py (hunk 1 above)
Comment (abelikt) [reply tests/PySys/local_mqtt_pub_sub/run.py:32]: That probably depends heavily on the environment and when the scheduler decides what to run. It failed only in 1% of the cases. 100ms is maybe a bit long but has no real impact on the test duration.

**Pull request merged**